odoo/tools/view_validation.py

```diff
@@ -25,6 +30,32 @@ def decorator(fn):
     return decorator
 
 
+def _relaxng(view_type):
+    _relaxng_validator = None
+    with tools.file_open(os.path.join('base', 'rng', '%s_view.rng' % view_type)) as frng:
+        try:
+            relaxng_doc = etree.parse(frng)
+            _relaxng_validator = etree.RelaxNG(relaxng_doc)
+        except Exception:
+            _logger.exception('Failed to load RelaxNG XML schema for views validation')
+    return _relaxng_validator
+
+
+@validate('calendar', 'diagram', 'gantt', 'graph', 'pivot', 'search', 'tree')
+def schema_valid(arch):
+    """ Get RNG validator and validate RNG file."""
+    validator = _relaxng_dict.get(arch.tag)
+    if not validator:
+        validator = _relaxng(arch.tag)
+        _relaxng_dict[arch.tag] = validator
```

Comment:
With the small refactoring I suggested above, those 4 lines can be simplified to:
```python
validator = relaxng(arch.tag)
```